### ISSUE TITLE
Should A+B Be Used for Training?

### ISSUE THREAD

**Issue by Chae97** [Devin-Pi/uncertainty-estimation-for-ssl#1]
Thank you for providing a great explanation of the code.

I have a question about the implemented loss function in the code.
Is the model currently being trained using only A in ce_kl_loss?
To reproduce the performance described in the paper, should we uncomment A + B and train the model accordingly?
Additionally, which value should be used: annealing_coef = 0.01, or the commented-out part that dynamically adjusts it?

![Image](https://github.com/user-attachments/assets/ebdd9ae2-1442-4d5b-bf1e-73d78a3f9588)

![Image](https://github.com/user-attachments/assets/2cb83f71-1768-4f2f-b87a-a40bdbdc2914)

**Comment by Devin-Pi** [comment]
> Thank you for providing a great explanation of the code.
> 
> I have a question about the implemented loss function in the code. Is the model currently being trained using only A in ce_kl_loss? To reproduce the performance described in the paper, should we uncomment A + B and train the model accordingly? Additionally, which value should be used: annealing_coef = 0.01, or the commented-out part that dynamically adjusts it?
> 
> ![Image](https://github.com/user-attachments/assets/ebdd9ae2-1442-4d5b-bf1e-73d78a3f9588)
> 
> ![Image](https://github.com/user-attachments/assets/2cb83f71-1768-4f2f-b87a-a40bdbdc2914)

Hi, Chae!

To reproduce the performance described in the paper, using A is ok. Based on the experiments, using A +B will harm the performance of the sound source localization. Regarding the annealing coefficient, it should be dynamically adjusted. Please refer to this commented line for guidance: 'annealing_coef = min(1, global_step / annealing_step)'.

 I hope the above response addresses your question!

**Comment by Chae97** [comment]
Thank you for your kind response!

I would like to know which combination of loss functions and evidence calculation methods should be used to achieve the performance reported in the paper.

When I followed the tutorial in the repository, I was able to obtain performance similar to that reported in the paper for validation and test sets. However, when evaluating on LOCATA, the performance was significantly worse.

What could be the possible reasons for this discrepancy?

Thank you in advance for your help!

**Comment by Devin-Pi** [comment]
> Thank you for your kind response!
> 
> I would like to know which combination of loss functions and evidence calculation methods should be used to achieve the performance reported in the paper.
> 
> When I followed the tutorial in the repository, I was able to obtain performance similar to that reported in the paper for validation and test sets. However, when evaluating on LOCATA, the performance was significantly worse.
> 
> What could be the possible reasons for this discrepancy?
> 
> Thank you in advance for your help!

Hi, Chae!

For the loss, just using the edl_loss (only A) is enough. For the LOCATA, are all results or just some results terrible? Have you tried the trained checkpoint? The link is here https://drive.google.com/drive/folders/1Yn3kfe-RG_rK7jjXyd8BU5dMEddRTzW0?usp=sharing.
